created one endpoint that parses and persists data + tdd for it

dancivasi opened the pull request.

dancivasi committed:
created one endpoint that parses and persists data + tdd for it

adrianmoloce commented:
Again, please remove the .idea/  directory from git.

adrianmoloce commented on main.py:
```diff
@@ -0,0 +1,21 @@
+app = FastAPI()
+
+with open('events.json', 'r') as file:
+    events = file.read()
```
You could simply do json.loads(file.read()) here.

adrianmoloce commented on main.py:
```diff
@@ -0,0 +1,21 @@
+all_events = json.loads(events)
+
+
+@app.get('/get_event/{customer_id}')
```
This works for now, but it does provide an opportunity to mention REST apis which is a widely used approach to building API's. I suggest you study some documentation or the original paper.

With REST conventions this endpoint's path should point to the events resource, thus it could be `/events/{customer_id}`

You can see how this is a more meaningful name as you actually provide all events for a customer_id not just a single event. The function name should also reflect that.

adrianmoloce commented on tests/test_main.py:
```diff
@@ -0,0 +1,26 @@
+client = TestClient(app)
+
+
+def test_there_is_data_in_json_file():
```
We do not test that a file contains data, this is not a unit test. It tests not functionality.

adrianmoloce commented on tests/test_main.py:
```diff
@@ -0,0 +1,26 @@
+    assert len(data) > 0
+
+
+def test_get_event_with_valid_customer_id():
```
This unit test has a great structure however it's an issue that the test is not isolated. If the file doesn't exist or someone modifies this data the test will start failing. In this very specific scenario, you could take a look at mock_open and patch the file that's actually being used and replace it to return fixed testing data.

adrianmoloce commented on tests/test_main.py:
```diff
@@ -0,0 +1,26 @@
+        {"customer_id": 123, "event_type": "purchase", "timestamp": "25-10-2023T15:33:00", "email_id": 1234, "product_id": 357, "amount": 49.99}
+            ]
+
+def test_get_event_with_invalid_customer_id():
```
This test is good, you could also opt to send a status code of 204 No Content with the message, but this works as well. Add a newline to the end of this file.

dancivasi committed:
changed some reviews

amoloce committed:
fix: Fix response model for the events endpoint

amoloce committed:
fix: Fix mock_open to work.

  - Fixed mocking open by not loading the data at import time into
    all_events and instead creating a method

amoloce committed:
fix: Add comments for option 2 but stick with 1

amoloce committed:
Merge branch 'master' of github.com:dancivasi/offer-fit-integration into 1endpoint

amoloce committed:
Merge branch 'master' of github.com:dancivasi/offer-fit-integration into 1endpoint

amoloce committed:
Merge branch 'master' of github.com:dancivasi/offer-fit-integration into 1endpoint

amoloce committed:
feat: Add coverage config file to omit tests cov
